server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -112,7 +111,7 @@
  *
  * @opensearch.internal
  */
-public class Translog extends AbstractIndexShardComponent implements IndexShardComponent, Closeable {
+public abstract class Translog extends AbstractIndexShardComponent implements IndexShardComponent, Closeable {
```

Comment:
Going through the Translog class, it still is heavily tied to local filesystem. Is it possible to have abstractions such that we could have a Translog class which is agnostic to Local/Remote store and then a LocalTranslog and RemoteLocalTranslog so that we could have right abstractions when we want to become remote completely?

Reply:
This would be a big change and not needed urgently. Hence. we want to defer it as of now as we still write to local and then upload to remote . Downloading puts in local and then reads from local .